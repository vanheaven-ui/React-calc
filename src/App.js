import ButtonPanel from "./components/Buttonpanel";

function App() {
  const myText =
    "I have completed the project setup. Time to get building... Well I should remember to lint the code.";
  return (
    <div className="App" style={{ color: "red", fontSize: 64 }}>
      {myText}
    </div>
  );
}

export default App;
