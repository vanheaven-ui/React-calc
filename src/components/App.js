import React from 'react';
import ButtonPanel from './ButtonPanel';
import Display from './Display';
import calculate from '../logic/calculate';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      dataObj: {
        total: null,
        next: null,
        operation: null,
      },
    };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(buttonName) {
    const stateCopy = { ...this.state };
    const dataObjCopy = stateCopy.dataObj;
    const mutants = ['+/-', '%', '', ''];

    console.log((Number.isNaN(parseInt(buttonName, 10))));

    if (dataObjCopy.total === null && buttonName === '.') {
      dataObjCopy.total = '0';
    }

    if (Number.isNaN(parseInt(buttonName, 10)) && buttonName !== '=' && buttonName !== '.') {
      if (!mutants.includes(buttonName)) {
        dataObjCopy.operation = buttonName;
      }
    } else if (typeof (parseInt(buttonName, 10)) === 'number' && buttonName !== '=') {
      if (dataObjCopy.operation !== null) {
        if (dataObjCopy.next === null) {
          dataObjCopy.next = buttonName;
        } else {
          dataObjCopy.next += buttonName;
        }
      } else if (dataObjCopy.total === null) {
        dataObjCopy.total = buttonName;
      } else if (dataObjCopy.total !== null) {
        dataObjCopy.total += buttonName;
      }
    }
    this.setState({ dataObj: calculate(dataObjCopy, buttonName) });
    console.log(dataObjCopy, buttonName);
  }

  render() {
    const { dataObj } = this.state;
    const nxt = dataObj.next;
    const opr = dataObj.operation;
    console.log(dataObj.total, nxt, opr);
    return (
      <>
        <Display
          output={dataObj.total === null ? '0' : dataObj.total.toString()}
          nxt={nxt === null ? '' : nxt.toString()}
          opr={opr === null ? '' : opr.toString()}
        />
        <ButtonPanel clickHandler={this.handleClick} />
      </>
    );
  }
}

export default App;
