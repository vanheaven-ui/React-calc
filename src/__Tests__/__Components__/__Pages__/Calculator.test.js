import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import App from '../../../components/pages/Calculator';
import '@testing-library/jest-dom';

test('is rendered correctly', () => {
  const rootNode = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter><App /></BrowserRouter>, rootNode,
  );
});

test('doesnot change unexpectedly', () => {
  const appTree = renderer.create(
    <BrowserRouter><App /></BrowserRouter>,
  ).toJSON();
  expect(appTree).toMatchSnapshot();
});

test('renders App component', () => {
  render(<BrowserRouter><App /></BrowserRouter>);
  expect(screen.getByText('Math Magicians')).toBeInTheDocument();
});
