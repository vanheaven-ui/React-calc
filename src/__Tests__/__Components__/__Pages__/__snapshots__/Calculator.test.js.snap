// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`doesnot change unexpectedly 1`] = `
Array [
  <nav
    className="navbar"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h2>
        Math Magicians
      </h2>
    </a>
    <div
      className="links"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        href="/calc"
        onClick={[Function]}
      >
        Calculator
      </a>
      <a
        href="/quotes"
        onClick={[Function]}
      >
        Quotes
      </a>
    </div>
  </nav>,
  <article
    className="calc-section"
  >
    <div
      className="calc-history"
    >
      <h2>
        Mathematics is a daily vital!!!
      </h2>
      <button
        className="hist-btn"
        onClick={[Function]}
        type="submit"
      >
        Get History
      </button>
    </div>
    <div
      className="calc"
    >
      <div
        className="display-window"
        data-testid="div"
      >
        0
        
        
      </div>
      <div
        className="btn-panel"
      >
        <div
          className="btn-group"
        >
          <button
            className="grey-btn default-btn"
            onClick={[Function]}
            type="button"
          >
            AC
          </button>
          <button
            className="grey-btn default-btn"
            onClick={[Function]}
            type="button"
          >
            +/-
          </button>
          <button
            className="grey-btn default-btn"
            onClick={[Function]}
            type="button"
          >
            %
          </button>
          <button
            className="default-btn-color default-btn"
            onClick={[Function]}
            type="button"
          >
            รท
          </button>
        </div>
        <div
          className="btn-group"
        >
          <button
            className="grey-btn default-btn"
            onClick={[Function]}
            type="button"
          >
            7
          </button>
          <button
            className="grey-btn default-btn"
            onClick={[Function]}
            type="button"
          >
            8
          </button>
          <button
            className="grey-btn default-btn"
            onClick={[Function]}
            type="button"
          >
            9
          </button>
          <button
            className="default-btn-color default-btn"
            onClick={[Function]}
            type="button"
          >
            x
          </button>
        </div>
        <div
          className="btn-group"
        >
          <button
            className="grey-btn default-btn"
            onClick={[Function]}
            type="button"
          >
            4
          </button>
          <button
            className="grey-btn default-btn"
            onClick={[Function]}
            type="button"
          >
            5
          </button>
          <button
            className="grey-btn default-btn"
            onClick={[Function]}
            type="button"
          >
            6
          </button>
          <button
            className="default-btn-color default-btn"
            onClick={[Function]}
            type="button"
          >
            -
          </button>
        </div>
        <div
          className="btn-group"
        >
          <button
            className="grey-btn default-btn"
            onClick={[Function]}
            type="button"
          >
            1
          </button>
          <button
            className="grey-btn default-btn"
            onClick={[Function]}
            type="button"
          >
            2
          </button>
          <button
            className="grey-btn default-btn"
            onClick={[Function]}
            type="button"
          >
            3
          </button>
          <button
            className="default-btn-color default-btn"
            onClick={[Function]}
            type="button"
          >
            +
          </button>
        </div>
        <div
          className="btn-group"
        >
          <button
            className="grey-btn double-btn"
            onClick={[Function]}
            type="button"
          >
            0
          </button>
          <button
            className="grey-btn default-btn"
            onClick={[Function]}
            type="button"
          >
            .
          </button>
          <button
            className="default-btn-color default-btn"
            onClick={[Function]}
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </article>,
]
`;
